src/shims/foreign_items.rs

```diff
@@ -369,38 +372,63 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let size = this.read_scalar(size)?.to_machine_usize(this)?;
                 let align = this.read_scalar(align)?.to_machine_usize(this)?;
                 Self::check_alloc_request(size, align)?;
-                let ptr = this.memory.allocate(
-                    Size::from_bytes(size),
-                    Align::from_bytes(align).unwrap(),
-                    MiriMemoryKind::Rust.into(),
-                )?;
-                this.write_pointer(ptr, dest)?;
+
+                let global_alloc = this.lookup_exported_symbol(Symbol::intern("__rg_alloc"))?;
```

Comment:
This seems like the mere existence of a symbol with that name would already make us use it. That is not the same logic as what rustc itself uses, is it? If I define a function `__rg_alloc` but do not use the global_allocator attribute, I think it will still use the default allocator, will it not?

Reply:
You're right, rustc uses different logic. There's an `allocator_kind` query, but it returns `Option<AllocatorKind>` and I'm unsure in which cases it returns `None`. I'll see if it works here.

Reply:
It looks like it only returns None when there is no global allocator, which is exactly what we want. I've changed the pr to use this query.